Remove exclusivity support for Swift 3 mode.

atrick opened the pull request:
Remove the compiler support for exclusivity warnings.

Leave runtime support for exclusivity warnings in non-release builds
only for unit testing convenience.

Remove a test case that checked the warning log output.

Modify test cases that relied on successful compilation in the
presence of exclusivity violations.

atrick committed:
Remove exclusivity support for Swift 3 mode.

Remove the compiler support for exclusivity warnings.

Leave runtime support for exclusivity warnings in non-release builds
only for unit testing convenience.

Remove a test case that checked the warning log output.

Modify test cases that relied on successful compilation in the
presence of exclusivity violations.

Fixes: <rdar://problem/45146046> Remaining -swift-version 3 tests for exclusivity

atrick commented:
@swift-ci test.

atrick commented:
@swift-ci test source compatibility.

atrick commented:
Fixes <rdar://problem/45146046> Remaining -swift-version 3 tests for exclusivity.

atrick commented:
@swift-ci test.

atrick commented:
@swift-ci test source compatibility.

slavapestov commented:
Thanks!

atrick commented:
@swift-ci test.

atrick commented:
@swift-ci test source compatibility.

devincoughlin commented on include/swift/ABI/MetadataValues.h:
```diff
@@ -997,8 +997,10 @@ enum class ExclusivityFlags : uintptr_t {
   ActionMask       = 0x1,
 
-  // Downgrade exclusivity failures to a warning.
+#ifndef NDEBUG
```
One possibility to remove the need to expose this in the ABI header is to separate out the unit tests that abort into a separate testing tool (similar to swift-ide-test) and turn the unit tests into file check tests.

devincoughlin reviewed (approved):
Looks good to me! Some comments in line.

atrick committed:
Create RuntimeUnittest library for C++ runtime unit tests from lit.

Create a new RuntimeUnittest library alongside the other stdlib unit
tests so we can write C++ runtime unit tests callable from lit.

Move runtime exclusivity tests into the stdlib unittest library and
create lit tests so we can verify that the runtime crashes with an
error message.

atrick committed:
Remove runtime support for exclusivity warnings.

atrick commented:
@devincoughlin @rjmccall I removed the remaining runtime support for warnings and moved the runtime unit tests into a library.

atrick commented:
@swift-ci test.

atrick commented:
@swift-ci test linux platform.

atrick commented:
@swift-ci test.
